File: src/ios/framework/data_store.m
#include "framework/data_store.h"

void dataStoreGlobalInit(void){
}

void dataStoreCommit(const char *dataString){
    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    NSString *state = [NSString stringWithUTF8String:dataString];
    [defaults setObject:state forKey:@"state"];
    [defaults synchronize];
}

const char* dataStoreReload(){
  NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
    [defaults synchronize];
  NSString *s = [defaults stringForKey:@"state"];
    if(s){
        return strdup([s UTF8String]);
    }else{
        return 0;
    }
}

File: src/common/framework/data_store.nim
when not defined(DATA_STORE_H_VI96GFE5): 
  const 
    DATA_STORE_H_VI96GFE5* = true
  proc dataStoreGlobalInit*() {.importc: "dataStoreGlobalInit", 
                                header: "data_store.h".}
  proc dataStoreCommit*(dataString: cstring) {.importc: "dataStoreCommit", 
      header: "data_store.h".}
  proc dataStoreReload*(): cstring {.importc: "dataStoreReload", 
                                     header: "data_store.h".}

